### PR TITLE
add #ifdef __MPI in codes related to MPI

### PR DESCRIPTION
**Pull request by lyb9812**
Add #ifdef __MPI in dftu, exx , and other codes.
Change makefile for serial version.

### CODE REVIEW

File: source/Makefile [hunk 1]
```diff
@@ -122,8 +128,14 @@ ${OBJ_DIR}/%.o:%.cpp
 ${OBJ_DIR}/%.o:%.f
 	${FORTRAN} -c ${HONG} $< -o $@	 
 
+${OBJ_DIR_serial}/%.o:%.cpp
```
Comment (caic99) [inline source/Makefile:131]: What's the difference between `OBJ_DIR_serial` and `OBJ_DIR`? From source file list to downstream compiling commands, they are identical.

File: source/input_conv.cpp [hunk 2]
```diff
@@ -446,6 +448,7 @@ void Input_Conv::Convert(void)
 //----------------------------------------------------------
 // about exx, Peize Lin add 2018-06-20
 //----------------------------------------------------------
+#ifdef __MPI // liyuanbo 2022/2/23
```
Comment (caic99) [inline source/input_conv.cpp:451]: Date and author info can be accessed by `git blame`.

File: source/input_conv.cpp [hunk 3]
```diff
@@ -503,7 +506,7 @@ void Input_Conv::Convert(void)
 		Exx_Abfs::Jle::tolerence = INPUT.exx_opt_orb_tolerence;
 	}
 #endif
-
+#endif
```
Comment (caic99) [inline source/input_conv.cpp:509]: Please add comments for `#endif` for long-ranged `#ifdef` block: `#endif // __MPI`

File: source/Makefile (hunk 1 above)
Comment (lyb9812) [reply source/Makefile:131]: OBJ_DIR uses CPLUSPLUS_MPI to compile and OBJ_DIR_serial uses CPLUSPLUS to compile.

File: source/input_conv.cpp (hunk 2 above)
Comment (lyb9812) [reply source/input_conv.cpp:451]: Thank you for your advices.

File: source/Makefile (hunk 1 above)
Comment (caic99) [reply source/Makefile:131]: > OBJ_DIR uses CPLUSPLUS_MPI to compile and OBJ_DIR_serial uses CPLUSPLUS to compile.

However, the source files between `FP_OBJS` and `FP_OBJS_serial` are of same(am I right? See line 71). I suggest using `${CPLUSPLUS}` with `${FP_OBJS}` but not `${FP_OBJS_serial}` in `serial` is enough, and no need to add `FP_OBJS_serial`. What do you think?

**Pull request merged**